client/material/button.tsx

```diff
@@ -27,17 +28,26 @@ export const Label = styled.span`
   white-space: nowrap;
 `
 
+export interface HotkeyProp {
+  keyCode: number
+  altKey?: boolean
+  shiftKey?: boolean
+  ctrlKey?: boolean
+}
+
 export interface ButtonStateProps {
+  ref?: React.ForwardedRef<HTMLButtonElement>
   disabled?: boolean
+  hotkey?: HotkeyProp
   onBlur?: (event: React.FocusEvent) => void
   onFocus?: (event: React.FocusEvent) => void
   onClick?: (event: React.MouseEvent) => void
   onDoubleClick?: (event: React.MouseEvent) => void
   onMouseDown?: (event: React.MouseEvent) => void
   onMouseEnter?: (event: React.MouseEvent) => void
   onMouseLeave?: (event: React.MouseEvent) => void
-  onKeyDown?: (event: React.KeyboardEvent) => void
-  onKeyUp?: (event: React.KeyboardEvent) => void
+  onKeyDown?: (event: KeyboardEvent) => void
```

Comment:
Honestly I think I would probably leave this as-is and continue attaching these handlers directly on the button. If you register these for the keylistener handler instead, the event is going to be handled in a different spot and thus things like `event.currentTarget` will not be the same. You're likely to break things that depend on that being a particular value (or just cause a lot of headaches for future people that expect the event to be handled at the button level)